### PR TITLE
Follow-up 2 to #1363 (update from spack dev 20241031): fix JEDI CI container builds for gcc and clang, and work towards fixing intel

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

This PR is a follow up to the update from spack develop in #1363. It removes the external `gmake` packages so that the new requirement for `gmake` to be one of `4.2.1` or `4.4.1` can be honored.

### Testing

- [x] `ubuntu-clang-mpich` with `jedi-ci` specs: https://github.com/JCSDA/spack-stack/actions/runs/11836736765/job/32982264182
- [x] `ubuntu-gcc-openmpi` with `jedi-ci` specs: https://github.com/JCSDA/spack-stack/actions/runs/11849699792/job/33023385023
- [ ] `ubuntu-intel-impi` with `jedi-ci` specs: https://github.com/JCSDA/spack-stack/actions/runs/11856641403/job/33043320658 - **Failed with an unrelated error that JCSDA needs to fix (also a result of the update from spack develop as of 2024/10/31:**
```
#18 ERROR: process "docker-shell cd /opt/spack-environment && spack env activate . && spack install --fail-fast && spack gc -y" did not complete successfully: exit code: 1
------
 > [builder 6/8] RUN cd /opt/spack-environment && spack env activate . && spack install --fail-fast && spack gc -y:
33.03 ==> Error: failed to concretize `fms@2024.02`, `g2@3.5.1`, `py-shapely@1.8.0`, `udunits@2.2.28`, ..., `py-pygithub@2.1.1` for the following reasons:
33.03      1. rust: Rust not compatible with Intel Classic compilers
33.03      2. rust: Rust not compatible with Intel Classic compilers
33.03         required because conflict is triggered when %intel 
33.03         required because conflict constraint  . You could consider setting `concretizer:unify` to `when_possible` or `false` to allow multiple versions of some packages.
```

### Applications affected

JEDI CI

### Systems affected

JEDI CI containers

### Dependencies

n/a

### Issue(s) addressed

n/a

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [ ] ~~All dependency PRs/issues have been resolved and this PR can be merged.~~

### CODE REVIEW

File: configs/containers/docker-ubuntu-clang-mpich.yaml [hunk 1]
```diff
@@ -59,11 +59,11 @@ spack:
...
-    gmake:
-      buildable: false
-      externals:
-      - spec: gmake@4.3
-        prefix: /usr
+    #gmake:
+    #  buildable: false
+    #  externals:
+    #  - spec: gmake@4.3
+    #    prefix: /usr
```
Comment (eap) [inline configs/containers/docker-ubuntu-clang-mpich.yaml]: Why comment here and delete elsewhere?

```suggestion
```

**Review by eap — APPROVED** [review]
Looks good, thanks for the fix.

**Comment by climbfuji** [comment]
Thanks for the quick review!

**Comment by climbfuji** [comment]
Created a follow-up issue for the remaining Intel error: https://github.com/JCSDA/spack-stack/issues/1383

**Review by stiggy87 — COMMENTED** [review]
Looks good to me!

**Pull request merged**